### PR TITLE
Fix log upload from openQA jobs in case of read-only cache directory

### PR DESCRIPTION
**Pull request by okurz**
We observed a problem that the os-autoinst command server received log
upload attempts but failed with

```
Mojo::Reactor::Poll: I/O watcher failed: Error in tempfile() using template /var/lib/openqa/cache/tmp/mojo.tmp.XXXXXXXXXXXXXXXX: Could not create temp file /var/lib/openqa/cache/tmp/mojo.tmp.mbmTdPv6NyxTOBTy: Permission denied at /usr/lib/perl5/5.26.1/File/Temp.pm line 1102.
```

Setting the environment variable MOJO_TMPDIR from the worker will mean
that also the os-autoinst command server effectively started by
the openQA worker will inherit

The original change was done at a time when the openQA worker itself
downloaded assets before the introduction of the separate cache service
which already handles that part correctly in OpenQA::Downloader so we
can simply remove the corresponding code without replacement.

Related progress issue: https://progress.opensuse.org/issues/119713

### CODE REVIEW

File: lib/OpenQA/Worker.pm [hunk 1]
```diff
@@ -579,9 +545,7 @@ sub _inform_webuis_before_stopping {
 }
 
 # stops the current job and (if there is one) and terminates the worker
-sub stop {
-    my ($self, $reason) = @_;
-
+sub stop ($self, $reason) {
```
Comment (Martchus) [inline lib/OpenQA/Worker.pm:548]: It should be ok to not pass a `$reason`.

File: lib/OpenQA/Worker.pm [hunk 2]
```diff
@@ -599,31 +563,23 @@ sub stop {
 }
 
 # stops the current job if there's one and it is running
-sub stop_current_job {
-    my ($self, $reason) = @_;
-
+sub stop_current_job ($self, $reason) {
```
Comment (Martchus) [inline lib/OpenQA/Worker.pm:566]: It should be ok to not pass a `$reason`.

**Review by Martchus — CHANGES_REQUESTED** [review]

File: lib/OpenQA/Worker.pm (hunk 1 above)
Comment (okurz) [reply lib/OpenQA/Worker.pm:548]: ok. I guess I will need to exclude further going signatures changes from this "fix" PR

**Comment by okurz** [comment]
Sorry, I need to open another PR https://github.com/os-autoinst/openQA/pull/4885

**Pull request closed without merging**